testng-core/src/main/java/org/testng/internal/invokers/ConfigInvoker.java

```diff
@@ -374,15 +375,30 @@ private void invokeConfigurationMethod(
         testResult.setStatus(ITestResult.SUCCESS);
         return;
       }
-      if (configurableInstance != null) {
-        MethodInvocationHelper.invokeConfigurable(
-            targetInstance, params, configurableInstance, method.getMethod(), testResult);
+      boolean willfullyIgnored = false;
+      boolean usesConfigurableInstance = configurableInstance != null;
+      if (usesConfigurableInstance) {
+        willfullyIgnored =
+            !MethodInvocationHelper.invokeConfigurable(
+                targetInstance, params, configurableInstance, method.getMethod(), testResult);
       } else {
         MethodInvocationHelper.invokeMethodConsideringTimeout(
             tm, method, targetInstance, params, testResult);
       }
-      testResult.setStatus(ITestResult.SUCCESS);
-    } catch (InvocationTargetException | IllegalAccessException ex) {
+      boolean testStatusRemainedUnchanged = testResult.isStatusCreatedOrStarted();
+      if (usesConfigurableInstance && willfullyIgnored && testStatusRemainedUnchanged) {
+        throw new ConfigurationNotInvokedException(
+            tm.getQualifiedName()
```

Comment:
You can build the message into ConfigurationNotInvokedException constructor where you can pass `tm`.

Reply:
Fixed for both the newly introduced exceptions.